README.md

```diff
@@ -197,25 +197,70 @@ need different templates for different distributions.
+`loop` list will allow you to add or override additional variables to
+specialize the values for platform and/or version.  Using the `when: item is
+file` test means that you do not have to provide all of the `vars/` files,
+only the ones you need.  For example, if every platform except Fedora uses
+`srv_name` for the service name, you can define `myrole_service: srv_name` in
+`vars/main.yml` then define `myrole_service: srv2_name` in `vars/Fedora.yml`.
+In cases where this would lead to duplicate vars files for similiar
+distibutions (e.g. CentOS 7 and RHEL 7), use symlinks to avoid the
+duplication.
+
+Platform specific tasks, however, are different.  You probably want to perform
+platform specific tasks once, for the most specific match.  In that case, use
+`with_first_found` with the file list in order of most specific to least
+specific, including a "default":
+```yaml
+- name: Perform platform/version specific tasks
+  include_tasks: "{{ item }}"
   with_first_found:
-    - "{{ ansible_distribution }}_{{ ansible_distribution_version }}.yml"
-    - "{{ ansible_distribution }}_{{ ansible_distribution_major_version }}.yml"
-    - "{{ ansible_distribution }}.yml"
-    - "{{ ansible_os_family }}.yml"
+    - files:
+        - "setup_{{ ansible_distribution }}_{{ ansible_distribution_version }}.yml"
+        - "setup_{{ ansible_distribution }}_{{ ansible_distribution_major_version }}.yml"
+        - "setup_{{ ansible_distribution }}.yml"
+        - "setup_{{ ansible_os_family }}.yml"
+        - "setup_default.yml"
+      paths:
+        - "{{ role_path }}/tasks"
 ```
+Then you would provide `tasks/setup_default.yml` to do the generic setup, and e.g. `tasks/setup_Fedora.yml` to do the Fedora specific setup.  The "setup_default.yml" is required in order to use `with_first_found`, which will give an error if no file is found.
 
-You can add `- default.yml` at the end to set the variables for unrecognized distributions from a common
-`default.yml` file (not to be confused with defaults for user-settable parameters, which go into
-`defaults/main.yml`). If this is not done, the role will fail for unrecognized distribution (this may or may
-not be intended). Put the distribution-specific variables into appropriate files under "vars/". Unfortunately,
-this would lead to duplicate vars files for similiar distibutions (e.g. CentOS 7 and RHEL 7). In cases such as
-these, use symlinks to avoid the duplication.
-
+If you want to have the "use first file found" semantics, but do not want to have to provide a default file, add `skip: true`:
+```yaml
+- name: Perform platform/version specific tasks
+  include_tasks: "{{ item }}"
+  with_first_found:
+    - files:
+        - "setup_{{ ansible_distribution }}_{{ ansible_distribution_version }}.yml"
+        - "setup_{{ ansible_distribution }}.yml"
+        - "setup_{{ ansible_os_family }}.yml"
+      paths:
+        - "{{ role_path }}/tasks"
+      skip: true
```

Comment:
Note that with_first_found is deprecated, and its replacement of using a lookup does not work with errors='ignore' for include_vars. See https://github.com/ansible/ansible/issues/58942 for more details.

Reply:
> Note that with_first_found is deprecated, and its replacement of using a lookup does not work with errors='ignore' for include_vars. See [ansible/ansible#58942](https://github.com/ansible/ansible/issues/58942) for more details.

The use of `with_<lookup>` syntax is not deprecated - see https://docs.ansible.com/ansible/latest/user_guide/playbooks_loops.html#
"Note
...
    We have not deprecated the use of with_<lookup> - that syntax will still be valid for the foreseeable future."

If you have different information, please let me know.  If you are referring to the error message emitted by the loop when using `skip: true`, I see no such message when using my simple test on Ansible 2.9.9:
```yaml
- name: Set version-specific variables
  include_vars: "{{ item }}"
  with_first_found:
    - files:
        - no_such_file.yml
      paths: "{{ role_path }}/vars"
      skip: true
```
Using `ANSIBLE_STDOUT_CALLBACK=debug ansible-playbook -vv role-var-test.yml` 
I see this output
```
TASK [role_b : Set version-specific variables] *********************************
task path: /home/rmeggins/ansible_sandbox/roles/role_b/tasks/main.yml:10
...

```
(that is, there is no output)
and I see this:
```
PLAY RECAP *********************************************************************
localhost                  : ok=7    changed=0    unreachable=0    failed=0    skipped=1    rescued=0    ignored=0
```
That is, the non existent file is silently skipped.

Reply:
He, I see https://github.com/ansible/ansible/commit/707e33793d683254a511cd1ec825df86a5121feb - I wonder whether that got ported back to 2.8 and therefore which range of versions will show the deprecated message. How confusing. :)

Reply:
> He, I see [ansible/ansible@707e337](https://github.com/ansible/ansible/commit/707e33793d683254a511cd1ec825df86a5121feb) - I wonder whether that got ported back to 2.8 and therefore which range of versions will show the deprecated message. How confusing. :)

Will use `lookup` based on guidance from Ansible team.